src/item_contents.cpp

```diff
@@ -0,0 +1,409 @@
+#include "item_contents.h"
+
+#include "generic_factory.h"
+#include "item.h"
+#include "item_pocket.h"
+#include "optional.h"
+#include "point.h"
+#include "units.h"
+
+namespace fake_item
+{
+static item_pocket none_pocket( nullptr );
+static pocket_data legacy_pocket_data;
+static item_pocket legacy_pocket( &legacy_pocket_data );
+} // namespace fake_item
+
+void item_contents::serialize( JsonOut &json ) const
+{
+    json.start_object();
+
+    json.member( "contents", contents );
+
+    json.end_object();
+}
+
+void item_contents::deserialize( JsonIn &jsin )
+{
+    JsonObject data = jsin.get_object();
+    optional( data, was_loaded, "contents", contents );
```

Comment:
```suggestion
    data.read( "contents", contents );
```
This format is used by the deserialize code in "src/savegame_json.cpp". The serialize and deserialize functions should go in that file as well.

And than you don't need the `was_loaded` member.